src/tests/users/signup.test.js

```diff
@@ -34,22 +151,71 @@ describe('signup', () => {
       query: `
         mutation {
           signup(record: {
-              fullName: "${fakeUser.fullName}",
-              email: "${fakeUser.email}",
-              password: "${fakeUser.password}"
+            firstName: "${fakeUser.firstName}"
+            lastName: "${fakeUser.lastName}"
+            email: "${fakeUser.email}"
+            password: "${fakeUser.password}"    
+            username: "${fakeUser.username}"
+          }) {
+            record {
+              firstName
+              lastName
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('errors')
+        expect(Array.isArray(res.body.errors)).toBe(true)
+      })
+      .expect(200)
+  })
+
+  it('should only create a new user if the username is unique', () => {
+    return request({
+      query: `
+        mutation {
+          signup(record: {
+            firstName: "${fakeUser.firstName}"
+            lastName: "${fakeUser.lastName}"
+            email: "${fakeUser.email}"
+            password: "${fakeUser.password}"    
+            username: "${fakeUser.username}"
           }) {
             record {
-              fullName
+              firstName
+              lastName
               email
             }
           }
         }
       `
     })
-      .expect(res => {
+      .expect((res) => {
         expect(res.body).toHaveProperty('errors')
         expect(Array.isArray(res.body.errors)).toBe(true)
       })
       .expect(200)
   })
+
+  it('user can be verified via verification code', () => {
```

Comment:
I don't understand this test. This is not, in any way, verifying anything but merely retrieving the verification code

Reply:
I'd rather say:

```javascript
mutation {
   verifyEmail(email: "test@test.com", code:"acfhkd2323") {
      record {
         email
         emailVerified
      }
  }
}
```
then create a custom mutation for `verifyEmail`, then create a custom static User model method `verifyEmail`.

And if this verification process is out of your scope, don't write the validation process and instead just assume the validation code will always be valid. That way, the interface (which you created) is tested, then whoever will write the method will use the same test and it will still pass, then another test for failure should be written at that point.

Reply:
added verify email